### PR TITLE
Fix tonemapper shader to correctly apply alpha channel

### LINKED ISSUE #40651: [4.0] The transparent_bg property in SubViewport seems to have no effect.
**Godot version:**  v4.0.dev.custom_build.3811fb919

**OS/device including version:** Arch Linux,  GeForce GTX 1070Q, Vulkan

**Issue description:** Enabling the "transparent_bg" property in SubViewport seems to have no effect. The 3D content within the viewport is rendered fine, but the underlying 2D content (ColorRect, TextureRect, etc.) is hidden below a black background. The expected outcome is to have the 3D content rendered on top of the 2D content.

**Note:** I'm aware 4.0 is anything but stable, so I'm wondering if this really is a bug or some problem with my machine. If anybody could test this, maybe on Windows, I'd highly appreciate it.

**Steps to reproduce:**
```
- Node
-- CanvasLayer
--- ColorRect
--- SubViewportContainer
---- SubViewport
----- Node3D
------ Camera3D
------ DirectionalLight
------ MeshInstance [Sphere]
```

### PR DESCRIPTION
**Pull request by Chaosus**
Changed a tone map shader to use an alpha channel from the input texture. 

This should fix a bug with `transparent_bg` in viewports:
image

Fix https://github.com/godotengine/godot/issues/40651

*Bugsquad edit: `master` version of https://github.com/godotengine/godot/pull/54585.*

### CODE REVIEW

File: servers/rendering/renderer_rd/forward_clustered/render_forward_clustered.cpp [hunk 1]
```diff
@@ -1523,13 +1523,13 @@ void RenderForwardClustered::_render_scene(RenderDataRD *p_render_data, const Co
...
 			Color cc = clear_color.srgb_to_linear();
-			if (using_separate_specular) {
+			if (using_separate_specular || render_buffer) {
 				cc.a = 0; //subsurf scatter must be 0
 			}
 			c.push_back(cc);
 
 			if (render_buffer) {
-				c.push_back(Color(0, 0, 0, 0));
+				c.push_back(Color(1, 1, 1, 1)); // Need a white color to correctly blending canvas item under viewport.
```
Comment (akien-mga) [inline servers/rendering/renderer_rd/forward_clustered/render_forward_clustered.cpp:1532]:
```suggestion
				c.push_back(Color(1, 1, 1, 1)); // Need a white color to correctly blend canvas item under viewport.
```

**Review by BastiaanOlij — APPROVED** [review]
This looks good to me and makes sense. I think there are other effects that may require a similar fix. I do know that DOF (I think) uses the alpha channel for weights and shouldn't overwrite the alpha of the end result.

**Comment by Chaosus** [comment]
@BastiaanOlij Yeah, I didn't touch other effects since I usually don't like to touch the things I've little to know ^^ to prevent accidental breakage.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**